test/end2end_test.go

```diff
@@ -3740,6 +3740,352 @@ func (s) TestClientStreaming_ReturnErrorAfterSendAndClose(t *testing.T) {
 	}
 }
 
+// Tests the behavior for server-side streaming when server calls RecvMsg twice.
+// Second call to RecvMsg should fail with Internal error.
+func (s) TestServerStreaming_ServerCallRecvMsgTwice(t *testing.T) {
+	lis, err := testutils.LocalTCPListener()
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer lis.Close()
+
+	s := grpc.NewServer()
+	serviceDesc := grpc.ServiceDesc{
```

Comment:
Do we need to fake the stream descriptors in this test? If not, we should use the test service client and server.